spark/src/main/java/org/apache/iceberg/spark/SparkUtil.java

```diff
@@ -38,6 +39,13 @@
   private SparkUtil() {
   }
 
+  public static Table toStaticTable(Table table) {
+    // TODO: consider setting some fields as null to reduce the size
```

Comment:
It may not make any difference. According to my preliminary tests, the size of a serialized static table is around 75226 bytes where almost 72000 bytes are taken by FileIO. I need to test it with tables that have a lot of columns and a number of specs and sort orders but it may not be worth it.

Reply:
How much better is the kryo serialization in the case of the Table compared to the Java serialization? Any numbers at hand?

Reply:
Not much, unfortunately. Spark cannot use Kryo on `FileIO` as it contains a Hadoop conf. Even though Spark may serialize the Table object using Kryo, we still use Java serialization for `FileIO` and that's the dominant part.

Reply:
Kryo doesn't work with Spark's `SerializableConfiguration`? I thought that's why we use the Spark one.

Reply:
It works but Spark uses Java serialization for Hadoop configs.

```
kryo.register(classOf[SerializableConfiguration], new KryoJavaSerializer())
```

By reusing `SerializableConfiguration`, we don't have to register our own custom Kryo serializer.

Reply:
Resolving as this is no longer relevant to this PR.